service-base/src/main/java/org/eclipse/hono/service/metric/DeviceConnectionDurationTracker.java

```diff
@@ -10,7 +10,7 @@
  *
  * SPDX-License-Identifier: EPL-2.0
  *******************************************************************************/
-package org.eclipse.hono.adapter.metric;
+package org.eclipse.hono.service.metric;
```

Comment:
It belongs there, yes. But keeping that class in the `adapter-base` module means having to refactor the `Metrics` interface, because the `incrementConnections`/`decrementConnections` method implementations in `MicrometerBasedMetrics` make use of the `DeviceConnectionDurationTracker` class.
Since this PR is already quite big, I think that should better be done in another PR.

Reply:
I see